api/all/build.gradle.kts

```diff
@@ -1,7 +1,6 @@
 plugins {
   id("otel.java-conventions")
   id("otel.publish-conventions")
-  id("java-test-fixtures")
```

Comment:
In hindsight, adding `java-test-fixtures` was overly hasty, since as shown in #6693 it impacts the `pom.xml` and publishes additional artifacts we didn't intend / account for. When we need to share test code in this repository, we simply add a `*:testing-internal` module and omit the `otel.publish-conventions`. I suspect this pattern was selected in the past after investigating test fixtures and seeing these types of side affects.

<img width="680" alt="Screenshot 2024-09-09 at 10 07 28 AM" src="https://github.com/user-attachments/assets/2c7cd49f-8d95-4959-9e86-328ea34e30f1">